### PR TITLE
duckdb: 0.5.1 -> 0.6.0

### PR DESCRIPTION
**Pull request by cpcloud**
###### Description of changes

Bump to 0.6.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cpcloud** [comment]
Result of `nixpkgs-review pr 201237` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>duckdb</li>
    <li>mycli</li>
    <li>python310Packages.duckdb</li>
    <li>python310Packages.duckdb-engine</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.sqlglot</li>
    <li>python39Packages.duckdb</li>
    <li>python39Packages.duckdb-engine</li>
    <li>python39Packages.ibis-framework</li>
    <li>python39Packages.sqlglot</li>
  </ul>
</details>

File: pkgs/development/python-modules/duckdb/default.nix [hunk 1]
```diff
@@ -13,13 +13,19 @@
...
+  # apply to the C++ library
   preConfigure = ''
     cd tools/pythonpkg
-    substituteInPlace setup.py --replace "multiprocessing.cpu_count()" "$NIX_BUILD_CORES"
+
+    # 1. let nix control build cores
+    # 2. unconstrain setuptools_scm version
+    substituteInPlace setup.py \
+      --replace "multiprocessing.cpu_count()" "$NIX_BUILD_CORES" \
+      --replace "setuptools_scm<7.0.0" "setuptools_scm"
```
Comment (lovesegfault) [inline pkgs/development/python-modules/duckdb/default.nix:28]: This should happen in `postPatch`, not `preConfigure`.

**Comment by lovesegfault** [comment]
Left a comment, LGTM to me otherwise, please squash as well

**Comment by cpcloud** [comment]
@lovesegfault, edited and squashed

File: pkgs/development/python-modules/duckdb/default.nix (hunk 1 above)
Comment (cpcloud) [reply pkgs/development/python-modules/duckdb/default.nix:28]: Done!

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 201237` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>duckdb</li>
    <li>mycli</li>
    <li>python310Packages.duckdb</li>
    <li>python310Packages.duckdb-engine</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.sqlglot</li>
    <li>python39Packages.duckdb</li>
    <li>python39Packages.duckdb-engine</li>
    <li>python39Packages.ibis-framework</li>
    <li>python39Packages.sqlglot</li>
  </ul>
</details>

**Pull request merged**